src/genericitems/producers.cpp

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Team Dissolve and contributors
+
+#include "genericitems/producers.h"
+#include "classes/neutronweights.h"
+#include "classes/partialset.h"
+#include "classes/xrayweights.h"
+#include "genericitems/list.h"
+#include "math/data1d.h"
+#include "math/data2d.h"
+#include "math/data3d.h"
+#include "math/histogram1d.h"
+#include "math/histogram2d.h"
+#include "math/histogram3d.h"
+#include <ios>
+
+GenericItemProducer::GenericItemProducer()
+{
+    // PODs
+    registerProducer<bool>("bool");
+    registerProducer<double>("double");
+    registerProducer<int>("int");
+
+    // stdlib
+    registerProducer<std::string>("std::string");
+    registerProducer<std::streampos>("streampos");
+    registerProducer<std::vector<double>>("std::vector<double>");
+
+    // Custom Classes
+    registerProducer<Array<double>>("Array<double>");
```

Comment:
```suggestion
    registerProducer<Array<double>>();
```

This change goes with the change I suggested to `registerProducer` in producers.cpp.  The same change would obviously be applied to all the following lines.

Reply:
Obviously not possible the the cross platform issues that we discussed.